dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresVectorStoreDbClient.cs

```diff
@@ -0,0 +1,260 @@
+
+namespace Microsoft.SemanticKernel.Connectors.Postgres;
+
+/// <summary>
+/// An implementation of a client for Postgres. This class is used to managing postgres database operations.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the <see cref="PostgresDbClient"/> class.
+/// </remarks>
+/// <param name="dataSource">Postgres data source.</param>
+/// <param name="schema">Schema of collection tables.</param>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities", Justification = "We need to build the full table name using schema and collection, it does not support parameterized passing.")]
+public class PostgresVectorStoreDbClient(NpgsqlDataSource dataSource, string schema = PostgresConstants.DefaultSchema) : IPostgresVectorStoreDbClient
+{
+    private readonly NpgsqlDataSource _dataSource = dataSource;
+    private readonly string _schema = schema;
+
+    private IPostgresVectorStoreCollectionSqlBuilder _sqlBuilder = new PostgresVectorStoreCollectionSqlBuilder();
+
+    /// <inheritdoc />
+    public async Task<bool> DoesTableExistsAsync(string tableName, CancellationToken cancellationToken = default)
+    {
+        NpgsqlConnection connection = await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+        await using (connection)
+        {
+            var commandInfo = this._sqlBuilder.BuildDoesTableExistCommand(this._schema, tableName);
+            using NpgsqlCommand cmd = commandInfo.ToNpgsqlCommand(connection);
+            using NpgsqlDataReader dataReader = await cmd.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
+            if (await dataReader.ReadAsync(cancellationToken).ConfigureAwait(false))
+            {
+                return dataReader.GetString(dataReader.GetOrdinal("table_name")) == tableName;
+            }
+
+            return false;
+        }
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<string> GetTablesAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        NpgsqlConnection connection = await this._dataSource.OpenConnectionAsync(cancellationToken).ConfigureAwait(false);
+
+        await using (connection)
+        {
+            var commandInfo = this._sqlBuilder.BuildGetTablesCommand(this._schema);
+            using NpgsqlCommand cmd = commandInfo.ToNpgsqlCommand(connection);
+            using NpgsqlDataReader dataReader = await cmd.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
+            while (await dataReader.ReadAsync(cancellationToken).ConfigureAwait(false))
+            {
+                yield return dataReader.GetString(dataReader.GetOrdinal("table_name"));
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task CreateTableAsync(string tableName, VectorStoreRecordDefinition recordDefinition, bool ifNotExists = true, CancellationToken cancellationToken = default)
```

Comment:
Consider only passing the properties, since that's all from the record definition that's required.